frontend/ui/widget/checkbutton.lua

```diff
@@ -109,39 +109,39 @@ function CheckButton:initCheckButton(checked)
 end
 
 function CheckButton:onTapCheckButton()
-    if self.enabled and self.callback then
-        if G_reader_settings:isFalse("flash_ui") then
-            self.callback()
+    if self.enabled then
+        if self.tap_input then
+            self:onInput(self.tap_input)
+        elseif type(self.tap_input_func) == "function" then
+            self:onInput(self.tap_input_func())
```

Comment:
Is this re-ordering required for any reason other than visually getting rid early of this use case ? :)

Reply:
There are no `tap_input`s anywhere currently, but as I understand from the original code, `tap_input` overrides a callback.

Reply:
I think the proposed logic is good, when there is an input, no callback.

Reply:
Toggling is needed when there is no tap_input, not depending on the callback - hence the order.

Reply:
I vaguely recall that the convoluted logic was there for a specific corner case, but damned if I remember what.

Also far too short on time to actually review this :/.

Reply:
(Possibly involving self.enabled being changed by the callback or something).

Reply:
The only change is that now flashing and toggling are done even when there is no callback.